### PR TITLE
tests: enable nested uc24

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This pr is the initial change to start executing nested tests in uc24

After this change it is required to start adding te missing fixes

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -264,6 +264,10 @@ backends:
                   image: ubuntu-2204-64-virt-enabled
                   storage: 25G
                   workers: 10
+            - ubuntu-24.04-64:
```
Comment (andrewphelpsj) [inline spread.yaml:267]: Missing an entry in `google-nested-dev` for 24

**Review by andrewphelpsj — APPROVED** [review]
Thank you!

**Review by ernestl — COMMENTED** [review]
Details missing causing static check fail. Are we specifically not adding it to save time?

**Comment by ernestl** [comment]
This is not related to nested, but nightly/upload-snapd-to-gce/task.yaml

This does not seem to cover core24. Can we not change function to and add ubuntu-<this part> to ubuntu-core-<here> tp ensure its always covered?

**Comment by ernestl** [comment]
It would be a nice feature to allow regex style to specify systems in tests e.g. `ubuntu-2[2-9].04-64`.

**Review by ernestl — APPROVED** [review]
Thanks. I double checked that all relevant core and manual nested tests was modified.

**Comment by bboozzoo** [comment]
static checks for a bunch of tasks due to missing details:
```
 Format checks failed for task tests/nested/core/core20-create-recovery/task.yaml
 - Key 'details' is mandatory
Format checks failed for task tests/nested/core/core20-degraded/task.yaml
 - Key 'details' is mandatory
Format checks failed for task tests/nested/core/core20-gadget-reseal/task.yaml
 - Key 'details' is mandatory
Format checks failed for task tests/nested/core/core20-kernel-failover/task.yaml
 - Key 'details' is mandatory
Format checks failed for task tests/nested/core/core20-kernel-reseal/task.yaml
 - Key 'details' is mandatory
Format checks failed for task tests/nested/core/coreconfig-services/task.yaml
```

**Pull request merged**